### ISSUE TITLE
DynamoDB+DynamoDBLocal putItem null attribute marshalling issue

### ISSUE THREAD

**Issue by dillaman** [aws/aws-sdk-java-v2#2543]
## Describe the bug
When using DynamoDBLocal with the DynamoDB v2 SDK, `putItem` requests will currently fail if passing a null attribute. The server will respond with `AwsServiceException: Supplied AttributeValue is empty`. Inspection shows that the v2 SDK has a properly formatted `PutItemRequest` with a `AttributeValue(NUL=true)` but when the DynamoDBLocalShared library attempts to convert the v2 `PutItemRequest` to a v1 `PutItemRequest` via JSON serialization/deserialization, the v1 `PutItemRequest` will have an uninitialized `AttributeValue`.

## Expected Behavior
Expect that the v2 SDK should be able to issue a `putItem` request with a NULL attribute value when using DynamoDBLocal.

## Current Behavior
DynamoDBLocal responds with a `AwsServiceException: Supplied AttributeValue is empty` error.

## Steps to Reproduce
Attempt to issue a `putItem` with a `String` attribute set to null. 

## Possible Solution
Fix the v2 -> v1 `PutItemRequest` marshalling code to properly handle `AttributeValue(NULL=true)` objects.

## Context
Cannot currently save null values using `DynamoDBLocal` and SDK v2.

## Your Environment

* AWS Java SDK version used: DynamoDBLocal 1.16.x
* JDK version used: JDK11
* Operating System and version: various

**Comment by debora-ito** [comment]
@dillaman can you provide a reproducible code?

**Comment by dillaman** [comment]
```
import com.amazonaws.services.dynamodbv2.local.embedded.DynamoDBEmbedded;
import org.junit.jupiter.api.Test;
import software.amazon.awssdk.enhanced.dynamodb.DynamoDbEnhancedClient;
import software.amazon.awssdk.enhanced.dynamodb.DynamoDbTable;
import software.amazon.awssdk.enhanced.dynamodb.TableSchema;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbAttribute;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;
import software.amazon.awssdk.services.dynamodb.DynamoDbClient;

import java.util.HashMap;
import java.util.Map;

public class NullAttributeTest {
    @DynamoDbBean
    public static class NullAttribute {
        @DynamoDbPartitionKey
        public String getHashKey() {
            return "hashKey";
        }
        public void setHashKey(String hashKey) {
        }

        @DynamoDbAttribute(value = "NullableMapAttribute")
        public Map<String, String> getNullableMapAttribute() {
            Map<String, String> result = new HashMap<>();
            result.put("key", null);
            return result;
        }
        public void setNullableMapAttribute(Map<String, String> nullableMapAttribute) {
        }
    }

    @Test
    void itShouldWork() {
        DynamoDbClient dbClient = DynamoDBEmbedded.create().dynamoDbClient();
        DynamoDbEnhancedClient dynamoDbEnhancedClient = DynamoDbEnhancedClient.builder().dynamoDbClient(dbClient).build();

        DynamoDbTable<NullAttribute> nullAttributeDynamoDbTable = dynamoDbEnhancedClient.table(
                "NullAttribute", TableSchema.fromBean(NullAttribute.class));
        nullAttributeDynamoDbTable.createTable();

        NullAttribute nullAttribute = new NullAttribute();
        nullAttributeDynamoDbTable.putItem(nullAttribute);
    }
}
```

**Comment by debora-ito** [comment]
@dillaman I'm sorry for the long silence in here, thank you for providing a repro code.

I can't reproduce the issue with the latest version of DynamoDB Local 1.18.0. The [docs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.UsageNotes.html) mention that DynamoDB Local supports the AWS SDK for Java V1 and V2, so maybe the issue was fixed since version 1.16.x.

Have you tested with the latest version?
